antarest/study/storage/rawstudy/model/filesystem/lazy_node.py

```diff
@@ -107,6 +108,25 @@ def delete(self, url: Optional[List[str]] = None) -> None:
         elif self.config.path.exists():
             self.config.path.unlink()
 
+    def infer_path(self) -> Path:
+        if self.get_link_path().exists():
+            return self.get_link_path()
+        elif self.config.path.exists():
+            return self.config.path
+        else:
+            raise ChildNotFoundError(
+                f"Neither link file {self.get_link_path} nor matrix file {self.config.path} exists"
+            )
+
+    def infer_target_path(self, is_link: bool) -> Path:
```

Comment:
I don't hink the notion of "target" makes sense here, it only makes sense in the contexte of a move or copy operation.

Instead of adding such introspection methods here, and have a complicated logic related to links in `update_binding_constraint`, we could offer `copy` and `move` methods, which will take care internally of the handling of links.

Reply:
Yeah, the design pattern at this point was poorly handled.